tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/contact/JCardObject.java

```diff
@@ -0,0 +1,67 @@
+package com.linagora.tmail.contact;
+
+import java.util.Optional;
+
+import jakarta.mail.internet.AddressException;
+
+import org.apache.james.core.MailAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.base.Preconditions;
+import com.linagora.tmail.james.jmap.contact.ContactFields;
+
+@JsonDeserialize(using = JCardObjectDeserializer.class)
+public record JCardObject(Optional<String> fnOpt, Optional<String> emailOpt) {
```

Comment:
Should we use MailAddress here or it's fine as is? And throw an exception at the deserializer.

Reply:
Of course using stronger types in the data model indeed is always better. If we think this can easily be achieved then we shall go for it.